### PR TITLE
DEVDOCS-0000: [update] Add new stencil-cli options

### PR DESCRIPTION
**Pull request by bc-0dp**
# [DEVDOCS-000]

## What changed?

* add new stencil-cli options to bundle and push
* add new section on theme optimization on source maps

## Release notes draft
Tools that provide automatic error reporting, session playback, and optimization suggestions can rely on source maps to give more readable insight. This update adds additional options to stencil-cli to optimize your themes performance with such tools.

ping @bc-traciporter

### CODE REVIEW

**Comment by jairo-bc** [comment]
LGTM

File: docs/storefront/stencil/deployment/performance-optimization.mdx [hunk 1]
```diff
@@ -132,6 +132,12 @@ After removing some unused bytes, minifying JavaScript files, and resizing an im
 
 Using Lighthouse is an important step in optimizing your site and improving the user experience. It is an easy to use best practice tip that provides powerful insights. 
 
+## Using Error Detections and Optimization Tools
```
Comment (bc-traciporter) [inline docs/storefront/stencil/deployment/performance-optimization.mdx:135]:
```suggestion
## Using error detections and optimization tools
```

File: docs/storefront/stencil/deployment/performance-optimization.mdx [hunk 2]
```diff
@@ -132,6 +132,12 @@ After removing some unused bytes, minifying JavaScript files, and resizing an im
...
+Tools that automatically record errors, provide session playback, and optimization suggestions often benefit from having access to the themes source maps in production.
```
Comment (bc-traciporter) [inline docs/storefront/stencil/deployment/performance-optimization.mdx:136]:
```suggestion
Tools that automatically record errors and provide session playback and optimization suggestions often benefit from having access to the themes' source maps in production. 
```

File: docs/storefront/stencil/deployment/performance-optimization.mdx [hunk 3]
```diff
@@ -132,6 +132,12 @@ After removing some unused bytes, minifying JavaScript files, and resizing an im
 
 Using Lighthouse is an important step in optimizing your site and improving the user experience. It is an easy to use best practice tip that provides powerful insights. 
 
+## Using Error Detections and Optimization Tools
+Tools that automatically record errors, provide session playback, and optimization suggestions often benefit from having access to the themes source maps in production. 
+They provide a way to translate between minified code, designed to decrease webpage load times, to the source code the developer writes.
```
Comment (bc-traciporter) [inline docs/storefront/stencil/deployment/performance-optimization.mdx:137]:
```suggestion
They provide a way to translate between minified code, designed to decrease webpage load times and the source code the developer writes.
```

File: docs/storefront/stencil/deployment/performance-optimization.mdx [hunk 4]
```diff
@@ -132,6 +132,12 @@ After removing some unused bytes, minifying JavaScript files, and resizing an im
...
+
+[Stencil-cli](/docs/storefront/stencil/cli/options-and-commands.mdx) provides convenient options for bundling the production ready theme with source maps included.
```
Comment (bc-traciporter) [inline docs/storefront/stencil/deployment/performance-optimization.mdx:139]:
```suggestion
[Stencil CLI](/docs/storefront/stencil/cli/options-and-commands) provides convenient options for bundling the production-ready theme with source maps included.
```

**Review by bc-traciporter — COMMENTED** [review]
I made some editorial comments for you to consider.

**Review by bc-traciporter — APPROVED** [review]
Approved

**Pull request merged**